minject/inject.py

```diff
@@ -91,6 +118,24 @@ def _is_type(key: "RegistryKey[T]") -> TypeGuard[Type[T]]:
     return isinstance(key, type)
 
 
+def _is_key_async(key: RegistryKey) -> bool:
+    """
+    Check whether a registry key is an async context manager
+    marked for initialization within the registry (@async_context).
+    """
+    if isinstance(key, str):
+        return False
+    elif isinstance(key, RegistryMetadata):
+        return key.is_async_context()
+    elif isinstance(key, type):
+        try:  # type: ignore
+            inject_metadata = object.__getattribute__(key, _INJECT_METADATA_ATTR)
+            return inject_metadata.is_async_context()  # type: ignore
+        except AttributeError:
+            return False
+    assert False, f"Unexpected key type: {key}"
```

Comment:
(nit) Maybe raise a `TypeError` instead of `AssertionError`, since the former is more specific?

Or, better still, change the last `elif` to an `else` and use `typing.assert_type(key, type)`?

Reply:
Much cleaner!